heft-plugins/heft-localization-typings-plugin/src/LocalizationTypingsPlugin.ts

```diff
@@ -0,0 +1,134 @@
+    });
+
+    taskSession.hooks.runIncremental.tapPromise(
+      PLUGIN_NAME,
+      async (runIncrementalOptions: IHeftTaskRunIncrementalHookOptions) => {
+        await this._runLocalizationTypingsGeneratorAsync(
+          taskSession,
+          slashNormalizedBuildFolderPath,
+          options,
+          runIncrementalOptions
+        );
+      }
+    );
+  }
+
+  private async _runLocalizationTypingsGeneratorAsync(
+    taskSession: IHeftTaskSession,
+    slashNormalizedBuildFolderPath: string,
+    {
+      exportAsDefault,
+      secondaryGeneratedTsFolders: secondaryGeneratedTsFoldersFromOptions,
+      exportAsDefaultDocumentationComment,
+      inferDefaultExportInterfaceNameFromFilename,
+      stringNamesToIgnore
+    }: ILocalizationTypingsPluginOptions | undefined = {},
+    runIncrementalOptions?: IHeftTaskRunIncrementalHookOptions
+  ): Promise<void> {
+    const logger: IScopedLogger = taskSession.logger;
+    const generatedTsFolderPath: string = `${slashNormalizedBuildFolderPath}/temp/loc-ts`;
+
+    let secondaryGeneratedTsFolders: string[] | undefined;
+    if (secondaryGeneratedTsFoldersFromOptions) {
+      secondaryGeneratedTsFolders = [];
+      for (const secondaryGeneratedTsFolder of secondaryGeneratedTsFoldersFromOptions) {
+        secondaryGeneratedTsFolders.push(`${slashNormalizedBuildFolderPath}/${secondaryGeneratedTsFolder}`);
+      }
+    }
+
+    const resxTypingsGenerator: TypingsGenerator = new TypingsGenerator({
+      srcFolder: `${slashNormalizedBuildFolderPath}/src`,
+      generatedTsFolder: generatedTsFolderPath,
+      terminal: logger.terminal,
+      ignoreString: stringNamesToIgnore
+        ? (stringName: string) => stringNamesToIgnore.includes(stringName)
+        : undefined,
+      exportAsDefault,
+      secondaryGeneratedTsFolders,
+      exportAsDefaultDocumentationComment,
+      inferDefaultExportInterfaceNameFromFilename
+    });
+
+    // If we have the incremental options, use them to determine which files to process.
+    // Otherwise, process all files. The typings generator also provides the file paths
+    // as relative paths from the sourceFolderPath.
+    let changedRelativeFilePaths: string[] | undefined;
+    if (runIncrementalOptions) {
+      changedRelativeFilePaths = [];
+      const relativeFilePaths: Map<string, IWatchedFileState> = await runIncrementalOptions.watchGlobAsync(
+        resxTypingsGenerator.inputFileGlob,
+        {
+          cwd: resxTypingsGenerator.sourceFolderPath,
+          ignore: Array.from(resxTypingsGenerator.ignoredFileGlobs),
+          absolute: false
+        }
+      );
+      for (const [relativeFilePath, { changed }] of relativeFilePaths) {
+        if (changed) {
+          changedRelativeFilePaths.push(relativeFilePath);
+        }
+      }
+      if (changedRelativeFilePaths.length === 0) {
+        return;
+      }
+    }
+
+    taskSession.logger.terminal.writeLine('Generating localization typings...');
```

Comment:
Nit: do we postfix these types of comments with `...` normally or just leave them without?

Reply:
This is what the sass typings generator does.